OsmAnd/src/net/osmand/plus/GPXUtilities.java

```diff
@@ -989,87 +1009,44 @@ public static GPXFile loadGPXFile(Context ctx, InputStream f) {
 					} else if (parse instanceof GPXExtensions && tag.equals("extensions")) {
 						extensionReadMode = true;
 					} else {
-						if (parse instanceof GPXFile) {
```

Comment:
You can discard this part of the change.
There is/was a slight bug in the original code which checked “tag” instead of “getName” but probably wasn’t that important.
To be clear - I don’t think I actually added any functionality in loadGPXFile - I just cleaned it up.  So, leave this change OUT.
Cheers
A



> On 20 Mar 2016, at 11:30 PM, vshcherb <notifications@github.com> wrote:
> 
> In OsmAnd/src/net/osmand/plus/GPXUtilities.java <https://github.com/osmandapp/Osmand/pull/2351#discussion_r56763597>:
> 
> > @@ -989,87 +1009,44 @@ public static GPXFile loadGPXFile(Context ctx, InputStream f) {
> >  					} else if (parse instanceof GPXExtensions && tag.equals("extensions")) {
> >  						extensionReadMode = true;
> >  					} else {
> > -						if (parse instanceof GPXFile) {
> This is quite a big change and because of the permutation of the lines. Will it be possible to reduce the diff so we can check semantically the changes rather than deep testing.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/osmandapp/Osmand/pull/2351/files/7c74c9581ed39b60f4356b973275e358636ba299#r56763597>